### ISSUE TITLE
Feature request: summary report

### ISSUE THREAD

**Issue by gemal** [palantir/tslint#2267]
### Bug Report

- __TSLint version__: 4.50
- __TypeScript version__:
- __Running TSLint via__: CLI

Could be nice to be able to see the files being linted like eslint can do with:
`DEBUG=eslint:* eslint .`

perhaps a `tslint --showfiles` option?
#### TypeScript code being linted

```ts
// code snippet
```

with `tslint.json` configuration:

```json

```

**Comment by gemal** [comment]
are you sure of this. Verbose only seem to show files with errors. I want to show every files that is linked, regardless of it has errors or not

**Comment by nchen63** [comment]
ah, sorry, I misunderstood the request.

I'm ok with the flag, but I'm not sure if printing to console without going through a formatter is the right approach. 

Perhaps we could add another function to IFormatter and formatters could optionally implement printing them in their own manner.

**Comment by gemal** [comment]
yes that sounds like a good idea

**Comment by gemal** [comment]
so another function besides:
`format(failures: RuleFailure[], fixes?: RuleFailure[]): string;`

like
`show(files: Files[]): string;`

??

I'm not a big expect here. Can you provide code or?

**Comment by nchen63** [comment]
hmm, how about `format(failures: RuleFailure[], fixes?: RuleFailure[], files?: string[]): string;`

**Comment by adidahiya** [comment]
I still don't see why you need this feature. Are you just verifying your glob expansion?

**Comment by gemal** [comment]
I use this in CI automation to make sure that all ts files were linted and that the glob and any ignores etc works

**Comment by adidahiya** [comment]
Given that we don't have https://github.com/palantir/tslint/issues/73... I don't think we need this implemented in TSLint itself. Can you verify your tslint file list config outside of the `tslint` command in CI?

**Comment by gemal** [comment]
I was using:
`tslint src/**/*.ts`

but apparently no files were linted
I should have used:
`tslint 'src/**/*.ts'`

so here a list of linted files would have been really nice :)

**Comment by chrismbarr** [comment]
What if it just had a final summary that said something like:

> TSLint finished linting 25 files, 3 have errors.

**Comment by gemal** [comment]
could be ok. a list of files would be better IMHO

**Comment by guilhermejcgois** [comment]
Some update about this? I would like at least some simple report like @chrismbarr sample...

**Comment by JoshuaKGoldberg** [comment]
Looks like this issue has morphed into a version of #3941: Feature request: summary report. Changing the title here to preserve the deeper discussion.

**Comment by mrseanryan** [comment]
>> I want to show every files that is linted, regardless of it has errors or not

@gemal to achieve this you could simply write a custom tslint rule, that console.logs each file it processes.

**Comment by JoshuaKGoldberg** [comment]
Perhaps relevant to the discussion: https://github.com/karfau/tslint-report

**Comment by karfau** [comment]
@JoshuaKGoldberg thx for brigning my attention to this issue.
As far as I understand the issue description/conversation the original intention was about the number of files that have been linted. Which is not what `tslint-report` provides.
Instead it is more related to the following topics:
> - Logging all known TSLint rules
> - Logging all enabled TSLint rules

That you added to this thread. But in that message it is not clear to me how you think the lined issues ask for such a thing.

For the sake of completeness: This rule set uses tslint-report and adds it's reports to it's repository:
- https://github.com/bettermarks/bm-tslint-rules/blob/master/tslint.report.sources.json
- https://github.com/bettermarks/bm-tslint-rules/blob/master/tslint.report.available.json
- https://github.com/bettermarks/bm-tslint-rules/blob/master/tslint.report.active.json

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
